### ISSUE TITLE
Use `note` tag "to inform mappers about non-obvious information about an element"

### ISSUE THREAD

**Issue by lgommans** [streetcomplete/StreetComplete#5794]
### Use case

I was approached by someone who got frustrated that one of his favorite shops was deleted twice now by people using StreetComplete. The problem is caused by the shop looking like a disused/vacant place outside of opening hours, and the opening hours aren't even fixed: they open at 6 in the morning and close when they're sold out, usually just after lunch time.

Normally, the [`note` tag](https://wiki.openstreetmap.org/wiki/Key%3Anote) can be used to signal to other mappers that there is something special about the object, per the wiki:

> The tag `note=*` is used:
> 
> - to document a general note about an object and
> - as a prefix, e.g. in `note:access=*`, to document why something has a specific `access=*` value.
> 
> [...] a note is generally used on nodes/ways/relations to describe real-world things like "the busstop is actually here but the signpost was knocked down in september 2009"

StreetComplete does not show this extra information to mappers who are editing the object.

### Proposed Solution

The simplest workaround would be to hide places with notes from the quest system, but I can't say that I would be a fan of this.

The best solution seems to me to display notes to the user somehow, before or while answering the quest. To avoid cluttering a bunch of quest UIs, *before* is probably the better option. It could show:

> This object has the following note attached: [...]  
> <kbd>Continue</kbd> <kbd>Hide quest</kbd> <kbd>Close</kbd>

(Where the <kbd>close</kbd> button just goes back to the map view.)

A middle ground would be to check for the presence of the `note` tag, and give the user a link to click so they can decide if they want to continue or not. For example, after answering a quest:

> This place has a note, please review it here: [openstreetmap.org/way/4443697](https://www.openstreetmap.org/way/4443697). Do you still want to apply this change?  
> <kbd>Apply</kbd> <kbd>Cancel</kbd>

Or before:

> This place has a note, please review it here: [openstreetmap.org/way/4443697](https://www.openstreetmap.org/way/4443697). Do you still want to answer this quest?  
> <kbd>Continue</kbd> <kbd>Cancel</kbd>

Further features might include letting the user move the `note` to the `description` tag if it describes the object rather than containing information relevant for mappers ([example](https://www.openstreetmap.org/node/1305725519/history/1)), preventing others from being shown the prompt unnecessarily, but that can be a later step. I would prioritise having some warning (just that a note exists is already enough for me) sooner rather than later, so I can prevent myself from making such a mistake again

**Comment by matkoniecz** [comment]
> The problem is caused by the shop looking like a disused/vacant place outside of opening hours, and the opening hours aren't even fixed

is it street vendor that pick up/closes its stall outside opening hours?

something like food truck that is simply not there outside its operating time? Or person who stands and sells stuff and leaves location when not selling things?

can you give a photo how it looks during opening hours and outside opening hours?

> was deleted twice now by people using StreetComplete

what kind of quests were asked there? was it the same person or two different one? can you link this OSM element?

**Comment by lgommans** [comment]
Hi Mateusz, it is a normal store in a building, but it gets boarded up when they're not open and no visible sign of an existing shop remains (also no business name on the mailbox, for example). Here is a picture while they are open, sent to me by the person who restored the shop: <https://up.picr.de/48506566pl.jpg>. I do not have a picture of the closed situation, as I did not think to take one while I was there. My visit was during the day though, so I would imagine that if the fridge with drinks had been visible while they're closed, I'd have noticed that it's stocked.

> what kind of quests were asked there? was it the same person or two different one? can you link this OSM element?

The quest being asked was whether there is indoor or outdoor seating. Link to object: <https://www.openstreetmap.org/node/867264057>, changeset in which I removed it erroneously: <https://www.openstreetmap.org/changeset/153896404>. I don't know who removed it last time, but it was not me so it was two different persons (as the Dutch saying goes: a donkey does not hit himself twice on the same stone... usually said when someone *does* make the same mistake twice :D)

**Comment by matkoniecz** [comment]
I hoped that https://wiki.openstreetmap.org/wiki/Tag:street_vendor%3Dyes would be applicable, but here it would be mistagging....

Though adding [outdoor_seating](https://wiki.openstreetmap.org/wiki/Key:outdoor%20seating?uselang=en) with proper value will at least shut up quest that resulted in deletion... (from photo it looks like outdoor_seating=no but it is not fully certain)

**Comment by kmpoppe** [comment]
I think this is a solid proposal. OsmAnd has a very intrusive way of displaying notes and fixmes, that's not a good example IMHO, but displaying a message in the edit screen that there is a note or a fixme sounds clever!

**Comment by lgommans** [comment]
Thanks @kmpoppe!

@matkoniecz Sure we can solve it for this particular one, but even once all the quests are solved, it will (or should) probably ask at some point if it's still there. Showing notes might solve it for the general case where an object has special circumstances to take into account. I guess another alternative could be to warn people when they tag something as nonexistent outside of its opening hours, but heeding the `note` field seems much more versatile

**Comment by matkoniecz** [comment]
> Sure we can solve it for this particular one, but even once all the quests are solved, it will (or should) probably ask at some point if it's still there.

Yes, that is more of a temporary solution.

**Comment by kjonosm** [comment]
For a current example, see also https://community.openstreetmap.org/t/aktuelle-offnungszeiten-nur-per-google-link/104684/16

**Comment by mnalis** [comment]
**TL;DR:** I think displaying note (if existing) if probably the best solution. See mockup below.

---

As an another example hitting me recently, there is (currently deleted) bakery in the middle of this picture https://www.openstreetmap.org/note/4283576 between "tobacco" shop on the right and ferry ticket sales on the left, currently behind rolled-down brown yellow planks.

To make things worse, it is open only during several months of tourist season, and even then only in the morning hours, which makes it quite likely it will get deleted again after it is re-added.

---

I absolutely agree that something should be done about it. 

While skipping all quests on elements that have `note=*` field is indeed too heavy, I initially though of maybe just disabling `Uh...` / `It does not exist...` answer on elements having `note`, thus forcing the user to leave a note.

But even that might be too heavy-handed, resulting in a lots of notes (there are over [3.9 million](https://taginfo.openstreetmap.org/keys/note#overview) `notes` on objects!), but, more importantly (as noted above), it does not solve the issue of `Is this still here?` quest.

So I do think that displaying the note to the user and letting them make an informed decision would probably be the best.

E.g. something like this mockup (I'd even prefer second one, to make it stand out, making it less likely to be ignored):
![small_SC_show_note_mockup](https://github.com/user-attachments/assets/1fbd4bca-28e0-485d-9fe6-edb0b1efc03a) ![small_SC_show_note_mockup2](https://github.com/user-attachments/assets/71972f16-4324-455f-8c97-9577d8dc397b)

**Comment by westnordost** [comment]
Hmm, how about the note is only displayed when the user selects that it doesn't exist? For example as a dialog that is shown before the "What's here now instead?" dialog.

**Comment by osmuser63783** [comment]
It would be helpful to show the opening hours in the "Is this still here?" quest, when they are mapped, or even to only show the quest during opening hours. (I know quite a few shops where it's impossible to tell when the shutters are down.)

**Comment by westnordost** [comment]
So, anyway, we have two options. Either 
- only show a note in a dialog when wanting to delete the place or 
- show the note always above the answer

Which way makes more sense depends on what's usually written in those notes. The former makes more sense if we want to guard against people deleting a place because it appears closed. The latter makes more sense if a note is expected to modify the user's answer in other cases, too.

**Comment by mnalis** [comment]
I'd personally probably prefer **always showing the note**, as it provides more utility for (I guess) about the same amount of work. 

And I guess it would not be too intrusive (e.g. out of 6.2 million shops, only some 32k have notes - which is [0.51%](https://taginfo.openstreetmap.org/keys/shop#combinations))

But I'm also fine if people prefer only showing note when trying to remove shop too. And it would probably work better for people who don't usually read what is on the screen unless it interrupts their flow - which might be a lot of them.

**Comment by kmpoppe** [comment]
I'm with @mnalis there. 

From the wiki for `note=*`: "A note is used to inform other mappers about non-obvious information about an element"[•••]

I'd argue that a SC surveyor is only able to observe the obvious information without involving a third party (i.e. questioning a person walking by), so providing those non-obvious information whereever possible and sensible would be my preferred way of doing it.

**Comment by lgommans** [comment]
I just noticed that, in 2023, [someone on the forum](https://community.openstreetmap.org/t/how-to-tag-unusable-sidewalks/97234/37) was already saying

> existence of `note` key might be checked programmatically by some editors like StreetComplete and potentially damaging quests might be skipped if it exists

So: tagging schemes have been [put on the wiki](https://wiki.openstreetmap.org/wiki/Sidewalks#Permanently_unusable_sidewalks) with this in mind, while it's not yet a thing :o

**Comment by kmpoppe** [comment]
I implemented a possible solution, see the linked PR. Notes are only shown for some quests that I thought might benefit greatly from such information.

**Comment by meiphoo7-Mae** [comment]
Do you really think this is going to work?
Let me tell a story: In my country there's an official agency that records outlines of buildings. Their information is publicly available. The local municipality provides the information but for some reason they seem to think certain secondary parts of buildings are not important.
So I found in my neighborhood a building (actually several buildings) with a wrong shape, which is pretty obvious from aerial images made by the same official agency, so I corrected it in OSM. A few days later someone had restored the original wrong shape, so I repaired it again and added a note not to change it again because the 'official' shape didn't represent the real situation and that I had surveyed it. Some time later it was reverted AGAIN. So I notified the official agency to update the shape according to their own images. They more or less penciled in the updated shape in their maps, so I updated it again in OSM and added a note NOT to change it and take look at aerial images. Guess what, someone has changed it for the third time to the OLD shape. For now I'm done with it and I'm not going to edit it a 4th time.
None of the actions by others (all of them were different users) were performed with SC/SCEE.

Bottomline:
1) Apparently not every user reads notes (probably most don't read notes).
2) Most users don't use SC/SCEE, so they are not going to be alerted that there's a note somewhere.
3) If someone really wants to add wrong information then there are a multitude of possibilities to do so.
4) Authors and users of software with the sole purpose of facilitating things can probably better stop to decide what is best for the community because it is simply pointless and a waste of their valuable time. See all of the above.

**Comment by kmpoppe** [comment]
I'm very sorry to hear that you had this unpleasant experience and that you now have - quite rightly - decided to give OSM The Finger about this edit because people were intent on destroying your work.

And while there are a multitude of ways this could have been dealt with (Changeset Comments, DWG), the very nature of OSM will always make this easy to happen.

What I'm saying is: the "time has been wasted" (while I don't think it was a waste because the solution we now have, see the PR associated with this issue, looks good!) and whether people will read the note or not is up to them. If it prevents the situation OP describes from happening only once, I'm happy. 

If people willingly change objects contrary to what notes say, no amount of prior effort will prevent them from doing so and it can only be reverted after the fact. I've accepted that a long time ago.

TL;DR: Thanks for your input and your effort, the implementation is done and we will see, if it changes anything.

K

**Comment by mnalis** [comment]
> TL;DR: Thanks for your input and your effort, the implementation is done and we will see, if it changes anything.

I agree. It will certainly help in **some** situations, but equally certainly will not help in **all** situations. As the saying goes: _"perfect is the enemy of the good"_. So improving situation even if the solution is not 100% foolproof is _still_ an improvement and should be cherished! :heart: 

> 2. Most users don't use SC/SCEE, so they are not going to be alerted that there's a note somewhere.

Well, most editors (with SC(EE) being an exception so far -- to be remedied with this PR!) **already do** show notes. Certainly popular ones like JOSM, iD, RapId, Vespucci, Every Door etc. do show  them.

> 3. If someone really wants to add wrong information then there are a multitude of possibilities to do so.

Sure, but that is not the point of this issue/PR. See [here](https://wiki.openstreetmap.org/wiki/Vandalism) how to handle vandalism.

---

As for your buildings issue @meiphoo7-Mae it is understandable why that was a downer, and I sympathize. Do not lose hope. I would've suggested to look at the changeset what sources were used by those users (and/or contact the authors via changeset discussions) and try to find root cause. If _different_ people are consistently adding _same_ wrong data, it means that same wrong data _must_ still be available somewhere. So finding where, and getting that source to update would've been the best course. Also, just [reverting](https://wiki.openstreetmap.org/wiki/Change_rollback) the problematic changesets would've been much easier and less painful than redrawing them from scratch, so I would consider that in the future when applicable.

**Comment by meiphoo7-Mae** [comment]
A little off topic:
I know about reverting, but at least the last unwanted change was part of a changeset with about 4900 edits nationwide (yes,... really!) and at least a part of the edits look valid to me. I'm not sure if a revert action will revert all of these edits or only the ones I'm interested in.

And on topic:
I'm not opposed against this new feature. I do actually read notes so this would make it easier.

**Comment by matkoniecz** [comment]
@meiphoo7-Mae 

The issue you mentions sounds worth acting on, if you would create thread about it at https://community.openstreetmap.org/ then feel free to ping me there.

>  None of the actions by others (all of them were different users) were performed with SC/SCEE.

so it is clearly offtopic to discuss on SC-specific issue tracker
